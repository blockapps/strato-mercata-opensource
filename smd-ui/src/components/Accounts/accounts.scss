.accounts-margin-top {
  bottom: 10px;
  padding-right: 7px;
  padding-left: 10px;
  overflow-y: auto;
  top: 170px;
}

.main-div {
  overflow-y: auto;
  margin-top: 40px;
  height: 70vh;
}

.main-div .smd-margin-8:first-of-type {
  margin-top: 0px;
}

.main-div::-webkit-scrollbar {
  width: 10px;
}

.account-details {
  margin-top: 40px;
  overflow-y: auto;
  height: 70vh;
  overflow-x: hidden;
}

.account-details::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
  border-radius: 4px;
  background: #2F424D; 
}
 
::-webkit-scrollbar-thumb {
  border-radius: 4px; 
  background: #F4F8FA; 
}

::-webkit-scrollbar-thumb:hover {
  background: #BBCBD4; 
}

.send-tokens-dialog .chain-field {
  width: 100%;
}