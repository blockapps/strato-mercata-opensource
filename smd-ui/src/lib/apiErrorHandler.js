export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
