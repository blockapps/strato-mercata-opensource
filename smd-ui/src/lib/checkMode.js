import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}