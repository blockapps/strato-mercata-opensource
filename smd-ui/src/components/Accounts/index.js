import React, { Component } from 'react';
import {
  fetchAccounts,
  fetchOauthAccounts,
} from './accounts.actions';
import mixpanelWrapper from '../../lib/mixpanelWrapper';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Tour from '../Tour';
import BlocAccounts from './components/BlocAccounts';
import OauthAccounts from './components/OauthAccounts';
import './accounts.css';
import { isOauthEnabled } from '../../lib/checkMode';
import ReactGA from 'react-ga4';

const tourSteps = [/* {
    title: 'Create User',
    text: 'Create a user here',
    selector: '#accounts-create-user-button',
    position: 'bottom', type: 'hover',
    isFixed: true,
  }, ) */
  {
    title: 'Upload a Smart Contract',
    text: 'Drag and drop a <strong>.sol</strong> file, and you will be able to manage your ' +
      'Smart Contract from within the STRATO dashboard.',
    /* text: '<div class="inline-code-sample">contract RentSplit {<br>address <strong>Roommate 1</strong>;<br><strong>Roommate 2</strong>;<br><strong>Roommate 3</strong>;<br>mapping (address => uint) RentSplit;<br></div>', */
    selector: '#contracts',
    position: 'bottom',
    isFixed: true
  }
];

class Accounts extends Component {

  constructor() {
    super()
    this.state = {
      selected: 0
    }
  }

  componentDidMount() {
    if (isOauthEnabled()) {
      this.props.fetchOauthAccounts();
    } else {
      this.props.fetchAccounts(true, true, this.props.selectedChain);
    }
    mixpanelWrapper.track('accounts_page_load');
    ReactGA.send({hitType: "pageview", page: "/accounts", title: "Accounts"});
  }

  render() {
    return (
      <div className="container-fluid pt-dark">
        <Tour
          name="accounts"
          steps={tourSteps}
          finalStepSelector='#contracts'
          nextPage='chains' />
        {isOauthEnabled() ? <OauthAccounts /> : <BlocAccounts />}
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {};
}

export default withRouter(
  connect(mapStateToProps,
    {
      fetchOauthAccounts,
      fetchAccounts
    }
  )(Accounts));
