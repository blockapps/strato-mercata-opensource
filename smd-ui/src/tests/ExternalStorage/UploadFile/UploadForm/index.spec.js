import React from 'react';
import UploadForm, { mapStateToProps } from '../../../../components/ExternalStorage/UploadFile/UploadForm';
import { createStore, combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import * as checkMode from '../../../../lib/checkMode';
import { accountsMock, indexAccountsMock } from '../../../Accounts/accountsMock';
import { mockFormData } from '../mockUpload';

describe('UploadFile: index', () => {
  let store

  beforeEach(() => {
    store = createStore(combineReducers({ form: formReducer }));
  });

  describe('render component (Oauth mode)', () => {
    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(false);
    })

    test('without values', () => {
      const props = {
        accounts: accountsMock,
        username: null,
        isLoading: false,
        initialValues: {
          username: null,
          address: null
        },
        uploadFileRequest: jest.fn(),
        reset: jest.fn(),
        changeUsername: jest.fn(),
        fetchUserAddresses: jest.fn(),
        closeUploadModal: jest.fn(),
        handleSubmit: jest.fn(),
        store: store
      }

      const wrapper = shallow(
        <UploadForm.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        accounts: indexAccountsMock,
        username: 'User_1177_79118',
        isLoading: false,
        initialValues: {
          username: 'User_1177_79118',
          address: '33c02a81e677ea493aace735f0a6b44cfa18f44e'
        },
        uploadFileRequest: jest.fn(),
        reset: jest.fn(),
        changeUsername: jest.fn(),
        fetchUserAddresses: jest.fn(),
        closeUploadModal: jest.fn(),
        handleSubmit: jest.fn(),
        store: store
      }

      const wrapper = shallow(
        <UploadForm.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

  });

  describe('render component (Non Oauth mode)', () => {
    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(true);
    })

    test('without values', () => {
      const props = {
        accounts: accountsMock,
        username: null,
        isLoading: false,
        initialValues: {
          username: null,
          address: null
        },
        uploadFileRequest: jest.fn(),
        reset: jest.fn(),
        changeUsername: jest.fn(),
        fetchUserAddresses: jest.fn(),
        closeUploadModal: jest.fn(),
        handleSubmit: jest.fn(),
        store: store
      }

      const wrapper = shallow(
        <UploadForm.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        accounts: indexAccountsMock,
        username: 'User_1177_79118',
        isLoading: false,
        initialValues: {
          username: 'User_1177_79118',
          address: '33c02a81e677ea493aace735f0a6b44cfa18f44e'
        },
        uploadFileRequest: jest.fn(),
        reset: jest.fn(),
        changeUsername: jest.fn(),
        fetchUserAddresses: jest.fn(),
        closeUploadModal: jest.fn(),
        handleSubmit: jest.fn(),
        store: store
      }

      const wrapper = shallow(
        <UploadForm.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

  });

  describe('Non oauth mode', () => {
    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(false);
    })

    test('Form input', () => {
      const props = {
        accounts: indexAccountsMock,
        username: 'User_1177_42489',
        isLoading: false,
        initialValues: {
          username: 'User_1177_42489',
          address: 'a421b5e3c34118266745dd8426a80b35513b2277'
        },
        uploadFileRequest: jest.fn(),
        reset: jest.fn(),
        changeUsername: jest.fn(),
        fetchUserAddresses: jest.fn(),
        closeUploadModal: jest.fn(),
        handleSubmit: jest.fn(),
        store: store
      }

      const wrapper = shallow(
        <UploadForm.WrappedComponent {...props} />
      ).dive().dive().dive();


      wrapper.find('Field').at(0).simulate('change', { target: { value: 'Supplier1' } });
      wrapper.find('Field').at(1).simulate('change', { target: { value: 'address' } });
      wrapper.find('Field').at(2).simulate('change', { target: { value: 'password' } });
      wrapper.find('Field').at(3).simulate('change', { target: { value: 'video' } });
      wrapper.find('Field').at(4).simulate('change', { target: { value: 's3' } });
      wrapper.find('Field').at(5).simulate('change', { target: { value: 'description' } });

      wrapper.find('Button').simulate('click');
      expect(props.handleSubmit).toHaveBeenCalled();
      expect(props.handleSubmit).toHaveBeenCalledTimes(1);
    });

    describe('onSubmit', () => {

      test('with values', () => {
        const props = {
          accounts: indexAccountsMock,
          username: 'User_1177_42489',
          isLoading: false,
          initialValues: {
            username: 'User_1177_42489',
            address: 'a421b5e3c34118266745dd8426a80b35513b2277'
          },
          uploadFileRequest: jest.fn(),
          reset: jest.fn(),
          changeUsername: jest.fn(),
          fetchUserAddresses: jest.fn(),
          closeUploadModal: jest.fn(),
          handleSubmit: jest.fn(),
          store: store
        }

        const wrapper = shallow(
          <UploadForm.WrappedComponent {...props} />
        ).dive().dive().dive();

        wrapper.instance().submit(mockFormData);
        expect(props.uploadFileRequest).toHaveBeenCalled();
        expect(props.uploadFileRequest).toHaveBeenCalledTimes(1);
      });

      test('without values', () => {
        const props = {
          accounts: indexAccountsMock,
          username: 'User_1177_42489',
          isLoading: false,
          initialValues: {
            username: 'User_1177_42489',
            address: 'a421b5e3c34118266745dd8426a80b35513b2277'
          },
          uploadFileRequest: jest.fn(),
          reset: jest.fn(),
          changeUsername: jest.fn(),
          fetchUserAddresses: jest.fn(),
          closeUploadModal: jest.fn(),
          handleSubmit: jest.fn(),
          store: store
        }

        const wrapper = shallow(
          <UploadForm.WrappedComponent {...props} />
        ).dive().dive().dive();

        wrapper.instance().submit({});
        expect(props.uploadFileRequest).not.toHaveBeenCalled();
        expect(props.uploadFileRequest).not.toHaveBeenCalledTimes(1);
      });

    });

  });

  describe('mapStateToProps', () => {

    test('without values', () => {
      const state = {
        accounts: {
          accounts: []
        },
        uploadFile: {
          username: null,
          isLoading: false
        },
        user: {
          oauthUser: {
            username: null,
            address: null
          }
        }
      }

      expect(mapStateToProps(state)).toMatchSnapshot();
    });

    test('with values', () => {
      const state = {
        accounts: {
          accounts: accountsMock
        },
        uploadFile: {
          username: 'Admin_1085_64667',
          isLoading: true
        },
        user: {
          oauthUser: {
            username: 'Admin_1085_64667',
            address: 'd2263b71c14010ff03d8f786670aba691b22b158'
          }
        }
      }

      expect(mapStateToProps(state)).toMatchSnapshot();
    });

  });

});