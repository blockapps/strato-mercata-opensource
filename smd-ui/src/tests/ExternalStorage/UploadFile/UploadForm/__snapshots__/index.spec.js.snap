// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadFile: index mapStateToProps with values 1`] = `
Object {
  "accounts": Array [
    "Admin_1085_64667",
    "Supplier1",
    "Supplier2",
    "Buyer1",
    "Buyer2",
    "Admin_1177_49507",
    "Admin_1177_39206",
    "Admin_1177_31203",
    "User_1177_14462",
    "123456789012345678901234567890123",
    "User_1177_26292",
    "User_1177_69645 ?#%!@*",
    "User_1177_59578",
    "User_1177_79118",
    "User_1177_59359",
    "User_1177_42489",
    "User_1177_37717",
    "User_1177_52307",
    "Buyer1177_18955",
    "Supplier1177_62701",
    "Admin_1301_86023",
    "Admin_1301_62816",
    "Supplier0_1301_46441",
    "Supplier1_1301_46441",
    "Supplier2_1301_46441",
    "tanuj",
  ],
  "initialValues": Object {
    "address": "d2263b71c14010ff03d8f786670aba691b22b158",
    "provider": "s3",
    "username": undefined,
  },
  "isLoading": true,
  "username": "Admin_1085_64667",
}
`;

exports[`UploadFile: index mapStateToProps without values 1`] = `
Object {
  "accounts": Array [],
  "initialValues": Object {
    "address": null,
    "provider": "s3",
    "username": undefined,
  },
  "isLoading": false,
  "username": null,
}
`;

exports[`UploadFile: index render component (Non Oauth mode) with values 1`] = `
"<div>
  <form>
    <div className=\\"pt-dialog-body upload-form\\">
      <div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Username
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"\\">
              <Field className=\\"pt-input\\" component=\\"select\\" name=\\"username\\" onChange={[Function]} required={true} disabled={true}>
                <option value=\\"User_1177_79118\\">
                  User_1177_79118
                </option>
                <option value=\\"Admin_1085_64667\\">
                  Admin_1085_64667
                </option>
                <option value=\\"Supplier1\\">
                  Supplier1
                </option>
                <option value=\\"Supplier2\\">
                  Supplier2
                </option>
                <option value=\\"Buyer1\\">
                  Buyer1
                </option>
                <option value=\\"Buyer2\\">
                  Buyer2
                </option>
                <option value=\\"Admin_1177_49507\\">
                  Admin_1177_49507
                </option>
                <option value=\\"Admin_1177_39206\\">
                  Admin_1177_39206
                </option>
                <option value=\\"Admin_1177_31203\\">
                  Admin_1177_31203
                </option>
                <option value=\\"User_1177_14462\\">
                  User_1177_14462
                </option>
                <option value=\\"123456789012345678901234567890123\\">
                  123456789012345678901234567890123
                </option>
                <option value=\\"User_1177_26292\\">
                  User_1177_26292
                </option>
                <option value=\\"User_1177_69645 ?#%!@*\\">
                  User_1177_69645 ?#%!@*
                </option>
                <option value=\\"User_1177_59578\\">
                  User_1177_59578
                </option>
                <option value=\\"User_1177_79118\\">
                  User_1177_79118
                </option>
                <option value=\\"User_1177_59359\\">
                  User_1177_59359
                </option>
                <option value=\\"User_1177_42489\\">
                  User_1177_42489
                </option>
                <option value=\\"User_1177_37717\\">
                  User_1177_37717
                </option>
                <option value=\\"User_1177_52307\\">
                  User_1177_52307
                </option>
                <option value=\\"Buyer1177_18955\\">
                  Buyer1177_18955
                </option>
                <option value=\\"Supplier1177_62701\\">
                  Supplier1177_62701
                </option>
                <option value=\\"Admin_1301_86023\\">
                  Admin_1301_86023
                </option>
                <option value=\\"Admin_1301_62816\\">
                  Admin_1301_62816
                </option>
                <option value=\\"Supplier0_1301_46441\\">
                  Supplier0_1301_46441
                </option>
                <option value=\\"Supplier1_1301_46441\\">
                  Supplier1_1301_46441
                </option>
                <option value=\\"Supplier2_1301_46441\\">
                  Supplier2_1301_46441
                </option>
              </Field>
            </div>
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Address
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"\\">
              <Field className=\\"pt-input\\" component=\\"select\\" name=\\"address\\" required={true} disabled={true}>
                <option value=\\"33c02a81e677ea493aace735f0a6b44cfa18f44e\\">
                  33c02a81e677ea493aace735f0a6b44cfa18f44e
                </option>
                <option value=\\"33c02a81e677ea493aace735f0a6b44cfa18f44e\\">
                  33c02a81e677ea493aace735f0a6b44cfa18f44e
                </option>
              </Field>
            </div>
            <br />
             
            <span className=\\"error-text\\" />
          </div>
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            File
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <Field id=\\"input-b\\" component={[Function]} className=\\"form-width pt-input\\" name=\\"content\\" dir=\\"auto\\" title=\\"Content\\" tabIndex=\\"4\\" required={true} />
           
          <br />
          <span className=\\"error-text\\" />
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            Provider
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <div className=\\"pt-select\\">
            <Field className=\\"pt-input\\" component=\\"select\\" name=\\"provider\\" tabIndex=\\"5\\" required={true}>
              <option value=\\"s3\\">
                 s3 
              </option>
            </Field>
             
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            Description
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <Field name=\\"description\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Description\\" className=\\"pt-input form-width\\" tabIndex=\\"6\\" required={true} />
           
          <br />
          <span className=\\"error-text\\" />
        </div>
      </div>
      <div className=\\"pt-dialog-footer upload-button\\">
        <div className=\\"pt-dialog-footer-actions button-center upload\\">
          <Blueprint.Button intent={0} onClick={[undefined]} disabled={false} text=\\"Upload\\" />
        </div>
      </div>
    </div>
  </form>
</div>"
`;

exports[`UploadFile: index render component (Non Oauth mode) without values 1`] = `
"<div>
  <form>
    <div className=\\"pt-dialog-body upload-form\\">
      <div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Username
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"\\">
              <Field className=\\"pt-input\\" component=\\"select\\" name=\\"username\\" onChange={[Function]} required={true} disabled={true}>
                <option value={{...}} />
                <option value=\\"0\\">
                  0
                </option>
                <option value=\\"1\\">
                  1
                </option>
                <option value=\\"2\\">
                  2
                </option>
                <option value=\\"3\\">
                  3
                </option>
                <option value=\\"4\\">
                  4
                </option>
                <option value=\\"5\\">
                  5
                </option>
                <option value=\\"6\\">
                  6
                </option>
                <option value=\\"7\\">
                  7
                </option>
                <option value=\\"8\\">
                  8
                </option>
                <option value=\\"9\\">
                  9
                </option>
                <option value=\\"10\\">
                  10
                </option>
                <option value=\\"11\\">
                  11
                </option>
                <option value=\\"12\\">
                  12
                </option>
                <option value=\\"13\\">
                  13
                </option>
                <option value=\\"14\\">
                  14
                </option>
                <option value=\\"15\\">
                  15
                </option>
                <option value=\\"16\\">
                  16
                </option>
                <option value=\\"17\\">
                  17
                </option>
                <option value=\\"18\\">
                  18
                </option>
                <option value=\\"19\\">
                  19
                </option>
                <option value=\\"20\\">
                  20
                </option>
                <option value=\\"21\\">
                  21
                </option>
                <option value=\\"22\\">
                  22
                </option>
                <option value=\\"23\\">
                  23
                </option>
                <option value=\\"24\\">
                  24
                </option>
                <option value=\\"25\\">
                  25
                </option>
                <option value=\\"length\\">
                  length
                </option>
              </Field>
            </div>
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Address
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"\\">
              <Field className=\\"pt-input\\" component=\\"select\\" name=\\"address\\" required={true} disabled={true}>
                <option value={{...}} />
              </Field>
            </div>
            <br />
             
            <span className=\\"error-text\\" />
          </div>
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            File
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <Field id=\\"input-b\\" component={[Function]} className=\\"form-width pt-input\\" name=\\"content\\" dir=\\"auto\\" title=\\"Content\\" tabIndex=\\"4\\" required={true} />
           
          <br />
          <span className=\\"error-text\\" />
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            Provider
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <div className=\\"pt-select\\">
            <Field className=\\"pt-input\\" component=\\"select\\" name=\\"provider\\" tabIndex=\\"5\\" required={true}>
              <option value=\\"s3\\">
                 s3 
              </option>
            </Field>
             
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            Description
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <Field name=\\"description\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Description\\" className=\\"pt-input form-width\\" tabIndex=\\"6\\" required={true} />
           
          <br />
          <span className=\\"error-text\\" />
        </div>
      </div>
      <div className=\\"pt-dialog-footer upload-button\\">
        <div className=\\"pt-dialog-footer-actions button-center upload\\">
          <Blueprint.Button intent={0} onClick={[undefined]} disabled={false} text=\\"Upload\\" />
        </div>
      </div>
    </div>
  </form>
</div>"
`;

exports[`UploadFile: index render component (Oauth mode) with values 1`] = `
"<div>
  <form>
    <div className=\\"pt-dialog-body upload-form\\">
      <div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Username
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"pt-select\\">
              <Field className=\\"pt-input\\" component=\\"select\\" name=\\"username\\" onChange={[Function]} required={true} disabled={false}>
                <option value={{...}} />
                <option value=\\"Admin_1085_64667\\">
                  Admin_1085_64667
                </option>
                <option value=\\"Supplier1\\">
                  Supplier1
                </option>
                <option value=\\"Supplier2\\">
                  Supplier2
                </option>
                <option value=\\"Buyer1\\">
                  Buyer1
                </option>
                <option value=\\"Buyer2\\">
                  Buyer2
                </option>
                <option value=\\"Admin_1177_49507\\">
                  Admin_1177_49507
                </option>
                <option value=\\"Admin_1177_39206\\">
                  Admin_1177_39206
                </option>
                <option value=\\"Admin_1177_31203\\">
                  Admin_1177_31203
                </option>
                <option value=\\"User_1177_14462\\">
                  User_1177_14462
                </option>
                <option value=\\"123456789012345678901234567890123\\">
                  123456789012345678901234567890123
                </option>
                <option value=\\"User_1177_26292\\">
                  User_1177_26292
                </option>
                <option value=\\"User_1177_69645 ?#%!@*\\">
                  User_1177_69645 ?#%!@*
                </option>
                <option value=\\"User_1177_59578\\">
                  User_1177_59578
                </option>
                <option value=\\"User_1177_79118\\">
                  User_1177_79118
                </option>
                <option value=\\"User_1177_59359\\">
                  User_1177_59359
                </option>
                <option value=\\"User_1177_42489\\">
                  User_1177_42489
                </option>
                <option value=\\"User_1177_37717\\">
                  User_1177_37717
                </option>
                <option value=\\"User_1177_52307\\">
                  User_1177_52307
                </option>
                <option value=\\"Buyer1177_18955\\">
                  Buyer1177_18955
                </option>
                <option value=\\"Supplier1177_62701\\">
                  Supplier1177_62701
                </option>
                <option value=\\"Admin_1301_86023\\">
                  Admin_1301_86023
                </option>
                <option value=\\"Admin_1301_62816\\">
                  Admin_1301_62816
                </option>
                <option value=\\"Supplier0_1301_46441\\">
                  Supplier0_1301_46441
                </option>
                <option value=\\"Supplier1_1301_46441\\">
                  Supplier1_1301_46441
                </option>
                <option value=\\"Supplier2_1301_46441\\">
                  Supplier2_1301_46441
                </option>
              </Field>
            </div>
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Address
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"pt-select\\">
              <Field className=\\"pt-input\\" component=\\"select\\" name=\\"address\\" required={true} disabled={false}>
                <option value={{...}} />
                <option value=\\"33c02a81e677ea493aace735f0a6b44cfa18f44e\\">
                  33c02a81e677ea493aace735f0a6b44cfa18f44e
                </option>
              </Field>
            </div>
            <br />
             
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Password
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <Field name=\\"password\\" component=\\"input\\" type=\\"password\\" placeholder=\\"Password\\" className=\\"pt-input form-width\\" tabIndex=\\"3\\" required={true} />
             
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            File
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <Field id=\\"input-b\\" component={[Function]} className=\\"form-width pt-input\\" name=\\"content\\" dir=\\"auto\\" title=\\"Content\\" tabIndex=\\"4\\" required={true} />
           
          <br />
          <span className=\\"error-text\\" />
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            Provider
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <div className=\\"pt-select\\">
            <Field className=\\"pt-input\\" component=\\"select\\" name=\\"provider\\" tabIndex=\\"5\\" required={true}>
              <option value=\\"s3\\">
                 s3 
              </option>
            </Field>
             
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            Description
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <Field name=\\"description\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Description\\" className=\\"pt-input form-width\\" tabIndex=\\"6\\" required={true} />
           
          <br />
          <span className=\\"error-text\\" />
        </div>
      </div>
      <div className=\\"pt-dialog-footer upload-button\\">
        <div className=\\"pt-dialog-footer-actions button-center upload\\">
          <Blueprint.Button intent={0} onClick={[undefined]} disabled={false} text=\\"Upload\\" />
        </div>
      </div>
    </div>
  </form>
</div>"
`;

exports[`UploadFile: index render component (Oauth mode) without values 1`] = `
"<div>
  <form>
    <div className=\\"pt-dialog-body upload-form\\">
      <div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Username
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"pt-select\\">
              <Field className=\\"pt-input\\" component=\\"select\\" name=\\"username\\" onChange={[Function]} required={true} disabled={false}>
                <option value={{...}} />
                <option value=\\"0\\">
                  0
                </option>
                <option value=\\"1\\">
                  1
                </option>
                <option value=\\"2\\">
                  2
                </option>
                <option value=\\"3\\">
                  3
                </option>
                <option value=\\"4\\">
                  4
                </option>
                <option value=\\"5\\">
                  5
                </option>
                <option value=\\"6\\">
                  6
                </option>
                <option value=\\"7\\">
                  7
                </option>
                <option value=\\"8\\">
                  8
                </option>
                <option value=\\"9\\">
                  9
                </option>
                <option value=\\"10\\">
                  10
                </option>
                <option value=\\"11\\">
                  11
                </option>
                <option value=\\"12\\">
                  12
                </option>
                <option value=\\"13\\">
                  13
                </option>
                <option value=\\"14\\">
                  14
                </option>
                <option value=\\"15\\">
                  15
                </option>
                <option value=\\"16\\">
                  16
                </option>
                <option value=\\"17\\">
                  17
                </option>
                <option value=\\"18\\">
                  18
                </option>
                <option value=\\"19\\">
                  19
                </option>
                <option value=\\"20\\">
                  20
                </option>
                <option value=\\"21\\">
                  21
                </option>
                <option value=\\"22\\">
                  22
                </option>
                <option value=\\"23\\">
                  23
                </option>
                <option value=\\"24\\">
                  24
                </option>
                <option value=\\"25\\">
                  25
                </option>
                <option value=\\"length\\">
                  length
                </option>
              </Field>
            </div>
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Address
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"pt-select\\">
              <Field className=\\"pt-input\\" component=\\"select\\" name=\\"address\\" required={true} disabled={false}>
                <option value={{...}} />
              </Field>
            </div>
            <br />
             
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Password
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <Field name=\\"password\\" component=\\"input\\" type=\\"password\\" placeholder=\\"Password\\" className=\\"pt-input form-width\\" tabIndex=\\"3\\" required={true} />
             
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            File
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <Field id=\\"input-b\\" component={[Function]} className=\\"form-width pt-input\\" name=\\"content\\" dir=\\"auto\\" title=\\"Content\\" tabIndex=\\"4\\" required={true} />
           
          <br />
          <span className=\\"error-text\\" />
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            Provider
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <div className=\\"pt-select\\">
            <Field className=\\"pt-input\\" component=\\"select\\" name=\\"provider\\" tabIndex=\\"5\\" required={true}>
              <option value=\\"s3\\">
                 s3 
              </option>
            </Field>
             
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
      </div>
      <div className=\\"row\\">
        <div className=\\"col-sm-3 text-right\\">
          <label className=\\"pt-label smd-pad-4\\">
            Description
          </label>
        </div>
        <div className=\\"col-sm-9 smd-pad-4\\">
          <Field name=\\"description\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Description\\" className=\\"pt-input form-width\\" tabIndex=\\"6\\" required={true} />
           
          <br />
          <span className=\\"error-text\\" />
        </div>
      </div>
      <div className=\\"pt-dialog-footer upload-button\\">
        <div className=\\"pt-dialog-footer-actions button-center upload\\">
          <Blueprint.Button intent={0} onClick={[undefined]} disabled={false} text=\\"Upload\\" />
        </div>
      </div>
    </div>
  </form>
</div>"
`;
